### ISSUE TITLE
undefined method `symbolize_keys' for Warden::JWTAuth:Module (NoMethodError

### ISSUE THREAD

**Issue by Mansh05** [waiting-for-dev/devise-jwt#160]
## Expected behavior
In rails 

in Gemfile
```
gem 'devise'
gem 'devise-jwt'

```
bundle install
rails s 
and it should be running as good as possible

In normal ruby
gem install devise-jwt
irb
> require 'devise'
true
> require 'devise-jwt'
true

## Actual behavior
1. In Rails

```
Traceback (most recent call last):
	19: from bin/rails:4:in `<main>'
	18: from bin/rails:4:in `require'
	17: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/commands.rb:18:in `<top (required)>'
	16: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/command.rb:46:in `invoke'
	15: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/command/base.rb:69:in `perform'
	14: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	13: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	12: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	11: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/commands/server/server_command.rb:142:in `perform'
	10: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/commands/server/server_command.rb:142:in `tap'
	 9: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/commands/server/server_command.rb:145:in `block in perform'
	 8: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/commands/server/server_command.rb:145:in `require'
	 7: from /Users/manish/selise/cnc-iam/config/application.rb:13:in `<top (required)>'
	 6: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler.rb:174:in `require'
	 5: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:58:in `require'
	 4: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:58:in `each'
	 3: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:69:in `block in require'
	 2: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:69:in `each'
	 1: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:74:in `block (2 levels) in require'
/Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:74:in `require': cannot load such file -- devise-jwt (LoadError)
	36: from bin/rails:4:in `<main>'
	35: from bin/rails:4:in `require'
	34: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/commands.rb:18:in `<top (required)>'
	33: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/command.rb:46:in `invoke'
	32: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/command/base.rb:69:in `perform'
	31: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/thor-1.0.1/lib/thor.rb:392:in `dispatch'
	30: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/thor-1.0.1/lib/thor/invocation.rb:127:in `invoke_command'
	29: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/thor-1.0.1/lib/thor/command.rb:27:in `run'
	28: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/commands/server/server_command.rb:142:in `perform'
	27: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/commands/server/server_command.rb:142:in `tap'
	26: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/commands/server/server_command.rb:145:in `block in perform'
	25: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/railties-5.2.4.1/lib/rails/commands/server/server_command.rb:145:in `require'
	24: from /Users/manish/selise/cnc-iam/config/application.rb:13:in `<top (required)>'
	23: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler.rb:174:in `require'
	22: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:58:in `require'
	21: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:58:in `each'
	20: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:65:in `block in require'
	19: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:88:in `rescue in block in require'
	18: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/bundler-2.1.4/lib/bundler/runtime.rb:88:in `require'
	17: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/devise-jwt-0.6.0/lib/devise/jwt.rb:6:in `<top (required)>'
	16: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/activesupport-5.2.4.1/lib/active_support/dependencies.rb:291:in `require'
	15: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/activesupport-5.2.4.1/lib/active_support/dependencies.rb:257:in `load_dependency'
	14: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/activesupport-5.2.4.1/lib/active_support/dependencies.rb:291:in `block in require'
	13: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/activesupport-5.2.4.1/lib/active_support/dependencies.rb:291:in `require'
	12: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/warden-jwt_auth-0.4.0/lib/warden/jwt_auth.rb:9:in `<top (required)>'
	11: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/warden-jwt_auth-0.4.0/lib/warden/jwt_auth.rb:20:in `<module:Warden>'
	10: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/warden-jwt_auth-0.4.0/lib/warden/jwt_auth.rb:41:in `<module:JWTAuth>'
	 9: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/dry-configurable-0.11.0/lib/dry/configurable/class_methods.rb:39:in `setting'
	 8: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/dry-configurable-0.11.0/lib/dry/configurable/dsl.rb:58:in `setting'
	 7: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/dry-configurable-0.11.0/lib/dry/configurable/compiler.rb:23:in `visit'
	 6: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/dry-configurable-0.11.0/lib/dry/configurable/compiler.rb:23:in `public_send'
	 5: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/dry-configurable-0.11.0/lib/dry/configurable/compiler.rb:29:in `visit_constructor'
	 4: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/dry-configurable-0.11.0/lib/dry/configurable/setting.rb:81:in `with'
	 3: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/dry-configurable-0.11.0/lib/dry/configurable/setting.rb:81:in `new'
	 2: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/dry-configurable-0.11.0/lib/dry/configurable/setting.rb:66:in `initialize'
	 1: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/dry-configurable-0.11.0/lib/dry/configurable/setting.rb:115:in `set!'
/Users/manish/.rvm/gems/ruby-2.6.4/gems/warden-jwt_auth-0.4.0/lib/warden/jwt_auth.rb:43:in `block in <module:JWTAuth>': undefined method `symbolize_keys' for Warden::JWTAuth:Module (NoMethodError)
```

2. In ruby

```
gem install devise-jwt
Successfully installed devise-jwt-0.6.0
Parsing documentation for devise-jwt-0.6.0
Done installing documentation for devise-jwt after 0 seconds
1 gem installed

2.6.4 :001 > require 'devise'
 => true 
2.6.4 :002 > require 'devise-jwt'
Traceback (most recent call last):
        9: from /Users/manish/.rvm/rubies/ruby-2.6.4/bin/irb:23:in `<main>'
        8: from /Users/manish/.rvm/rubies/ruby-2.6.4/bin/irb:23:in `load'
        7: from /Users/manish/.rvm/rubies/ruby-2.6.4/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        6: from (irb):2
        5: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/activesupport-6.0.2.1/lib/active_support/dependencies.rb:325:in `require'
        4: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/activesupport-6.0.2.1/lib/active_support/dependencies.rb:291:in `load_dependency'
        3: from /Users/manish/.rvm/gems/ruby-2.6.4/gems/activesupport-6.0.2.1/lib/active_support/dependencies.rb:325:in `block in require'
        2: from /Users/manish/.rvm/rubies/ruby-2.6.4/lib/ruby/site_ruby/2.6.0/rubygems/core_ext/kernel_require.rb:117:in `require'
        1: from /Users/manish/.rvm/rubies/ruby-2.6.4/lib/ruby/site_ruby/2.6.0/rubygems/core_ext/kernel_require.rb:117:in `require'
LoadError (cannot load such file -- devise-jwt)
2.6.4 :003 >
``` 

## Steps to Reproduce the Problem
A. 
1.  add devise and devise-jwt in rails
2. run bunlde install, runs good but when running rails s gives devise-jwt load error with warden symbolize keys error
3. Using pry, added a breakpoint in Warden::JWTAuth and gives the load error.

B.
1. install gem using gem install devise-jwt
2. run  irb
3. run require 'devise-jwt'

## Debugging information
- Version of `devise-jwt` in use: 0.6.0
- Version of `rails` in use: 5.2.4.1
- Output of `Devise::JWT.config`: uninitialized constant Devise::JWT
- Output of `Warden::JWTAuth.config`: {}
- Output of `Devise.mappings`: {}

**Comment by Mansh05** [comment]
The error comes when we upgrade the gem dry-configurator from 0.9.0 to 0.11.0 which comes with dry/equilizer. So make sure to keep your dry-configurator below 0.9.0

**Comment by Sartori-RIA** [comment]
same problem here

**Comment by pohl989** [comment]
@Mansh05 Thank you! I added this to my gemfile and everything is working now

`gem 'dry-configurable', '0.9.0'`

**Comment by serghost** [comment]
A few more related errors for future seekers:

`#<ArgumentError: +mappings+ is not a setting name>`
`+secret+ is not a setting name`

Also solved by downgrade to 0.9.0

**Comment by waiting-for-dev** [comment]
For now lock you dry-configurable version to 0.9. I'll take a look as soon as possible.

**Comment by waiting-for-dev** [comment]
Duplicate of #169. The issue has been fixed, check there for details.

**Comment by waiting-for-dev** [comment]
As it's taking longer than expected to be fixed upstream in dry-configurable, I have released a new version of warden-jwt_auth 0.4.2 which locks itself dry-configurable version. This way, once it's updated there I'll just release a new version and it'll be transparent for the user.

So, to fix it when using devise-jwt, just run `bundle update --source warden-jwt_auth`
